src/toolbar/toolbar-title.jsx

```diff
@@ -14,6 +14,10 @@ function getStyles(props, state) {
       fontSize: toolbar.titleFontSize,
       display: 'inline-block',
       position: 'relative',
+      textOverflow: 'ellipsis',
+      whiteSpace: 'nowrap',
+      overflow: 'hidden',
+      width: (props.text.length > 40) ? 150 : 'inherit',
```

Comment:
Yes we do for the long text to be converted to ellipses. Since the parent div has a width of 100%. I tried it without the width but it doesn't work. 

The width component actually depends on the use-case though. I have put a generic width statement which is at least a bit scalable.

Please find the explanation on the issue page: https://github.com/callemall/material-ui/issues/3192